server/core/runtime/state_rm_step_runner.go

```diff
@@ -0,0 +1,38 @@
+package runtime
+
+import (
+	"os"
+	"path/filepath"
+
+	version "github.com/hashicorp/go-version"
+	"github.com/runatlantis/atlantis/server/events/command"
+)
+
+type StateRmStepRunner struct {
+	TerraformExecutor TerraformExec
+	DefaultTFVersion  *version.Version
+}
+
+func (p *StateRmStepRunner) Run(ctx command.ProjectContext, extraArgs []string, path string, envs map[string]string) (string, error) {
+	tfVersion := p.DefaultTFVersion
+	if ctx.TerraformVersion != nil {
+		tfVersion = ctx.TerraformVersion
+	}
+
+	stateRmCmd := []string{"state", "rm"}
+	stateRmCmd = append(stateRmCmd, extraArgs...)
+	stateRmCmd = append(stateRmCmd, ctx.EscapedCommentArgs...)
+	out, err := p.TerraformExecutor.RunCommandWithVersion(ctx, filepath.Clean(path), stateRmCmd, envs, tfVersion, ctx.Workspace)
+
+	// If the state rm was successful and a plan file exists, delete the plan.
+	planPath := filepath.Join(path, GetPlanFilename(ctx.Workspace, ctx.ProjectName))
+	if err == nil {
+		if _, planPathErr := os.Stat(planPath); !os.IsNotExist(planPathErr) {
```

Comment:
Can we combine the if block on line 29 and 30?

Reply:
we can do `if _, planPathErr := os.Stat(planPath); err == nil && !os.IsNotExist(planPathErr) {` but it is better to check `err == nil` explicitly to figure out this logic for a successful process.

Reply:
Sounds reasonable